src/infinite-scroll.js

```diff
@@ -108,19 +108,23 @@ angular.module(MODULE_NAME, [])
         let timeout = null;
         let previous = 0;
 
-        function later() {
-          previous = new Date().getTime();
-          $interval.cancel(timeout);
+        function clear() {
+          if (timeout) {
+            $interval.cancel(timeout);
+          }
           timeout = null;
+        }
+        function later() {
+          clear();
```

Comment:
1. it was being "cleared" before but your right doesn't make sense.
2. This is just a small teak nothing changes in terms of ui to test.